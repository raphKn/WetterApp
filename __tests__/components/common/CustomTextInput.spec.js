import React from 'react';
import { shallow } from 'enzyme';
import CustomTextInputIos from '../../../src/components/common/CustomTextInput.ios';
import CustomTextInputAndroid from '../../../src/components/common/CustomTextInput.android';

const doNothing = () => {
};
describe('CustomTextInput ios', () => {
    it('renders in a snapshot', () => {
        const wrapper = shallow(
            <CustomTextInputIos onSubmitEditing={doNothing} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('handles changeText and submit correctly', () => {
        const mocksubmitEditing = jest.fn();
        const wrapper = shallow(
            <CustomTextInputIos onSubmitEditing={mocksubmitEditing} />);
        wrapper.props().children.props.onChangeText('inputValue');
        expect(wrapper.state('text')).toBe('inputValue');
        expect(mocksubmitEditing).toHaveBeenCalledTimes(0);
        wrapper.props().children.props.onSubmitEditing();
        expect(mocksubmitEditing).toHaveBeenCalledTimes(1);
        expect(mocksubmitEditing).toHaveBeenCalledWith('inputValue');
    });
});

describe('CustomTextInput android', () => {
    it('renders in a snapshot', () => {
        const wrapper = shallow(
            <CustomTextInputAndroid onSubmitEditing={doNothing} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('handles changeText and submit correctly', () => {
        const mocksubmitEditing = jest.fn();
        const wrapper = shallow(
            <CustomTextInputAndroid onSubmitEditing={mocksubmitEditing} />);
        wrapper.props().onChangeText('inputValue');
        expect(wrapper.state('text')).toBe('inputValue');
        expect(mocksubmitEditing).toHaveBeenCalledTimes(0);
        wrapper.props().onSubmitEditing();
        expect(mocksubmitEditing).toHaveBeenCalledTimes(1);
        expect(mocksubmitEditing).toHaveBeenCalledWith('inputValue');
    });
});
