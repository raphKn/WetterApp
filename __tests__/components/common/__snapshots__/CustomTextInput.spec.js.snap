// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTextInput android renders in a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CustomTextInput
    defaultText={null}
    onSubmitEditing={[Function]}
    returnKeyType={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "onChangeText": [Function],
      "onSubmitEditing": [Function],
      "returnKeyType": null,
      "style": Object {
        "color": "#212121",
      },
      "underlineColorAndroid": "#BDBDBD",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "onChangeText": [Function],
        "onSubmitEditing": [Function],
        "returnKeyType": null,
        "style": Object {
          "color": "#212121",
        },
        "underlineColorAndroid": "#BDBDBD",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CustomTextInput ios renders in a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CustomTextInput
    defaultText={null}
    onSubmitEditing={[Function]}
    returnKeyType={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TextInput
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        returnKeyType={null}
        style={
          Object {
            "color": "#212121",
          }
        }
        underlineColorAndroid="transparent"
      />,
      "style": Object {
        "borderBottomColor": "#BDBDBD",
        "borderBottomWidth": 1,
        "marginBottom": 8,
        "marginTop": 8,
        "paddingBottom": 8,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "onChangeText": [Function],
        "onSubmitEditing": [Function],
        "returnKeyType": null,
        "style": Object {
          "color": "#212121",
        },
        "underlineColorAndroid": "transparent",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          returnKeyType={null}
          style={
            Object {
              "color": "#212121",
            }
          }
          underlineColorAndroid="transparent"
        />,
        "style": Object {
          "borderBottomColor": "#BDBDBD",
          "borderBottomWidth": 1,
          "marginBottom": 8,
          "marginTop": 8,
          "paddingBottom": 8,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onChangeText": [Function],
          "onSubmitEditing": [Function],
          "returnKeyType": null,
          "style": Object {
            "color": "#212121",
          },
          "underlineColorAndroid": "transparent",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
